pkg/apis/networking/validation/validation.go

```diff
@@ -685,6 +698,27 @@ func allowInvalidWildcardHostRule(oldIngress *networking.Ingress) bool {
 	return false
 }
 
+func allowRelaxedServiceNameValidation(oldIngress *networking.Ingress) bool {
+	if oldIngress != nil {
+		for _, rule := range oldIngress.Spec.Rules {
+			for _, path := range rule.HTTP.Paths {
+				if path.Backend.Service == nil {
+					continue
+				}
+				serviceName := path.Backend.Service.Name
+				// If a name doesn't validate with NameIsDNS1035Label, but does validate with NameIsDNSLabel,
+				// then we allow it to be used as a Service name in an Ingress.
+				new := apimachineryvalidation.NameIsDNSLabel(serviceName, false)
+				old := apimachineryvalidation.NameIsDNS1035Label(serviceName, false)
+				if new == nil && old != nil {
+					return true
+				}
+			}
+		}
+	}
+	return false
```

Comment:
nit, it is a common practice to avoid excessive indentantion, in this case you can just do
```
if oldIngress == nil {
   return false
}
```

and you get the next section unindented

Reply:
Fixed